pkg/agent/openflow/feature_templates.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openflow
+
+import (
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+	"antrea.io/antrea/pkg/util/runtime"
+)
+
+func (c *featurePodConnectivity) getTemplate(protocol pipelineProtocol) *featureTemplate {
+	var template *featureTemplate
+	switch protocol {
+	case pipelineIP:
+		template = &featureTemplate{
+			stageTables: map[binding.StageID][]*FeatureTable{
+				binding.ClassifierStage: {
+					ClassifierTable,
+				},
+				binding.ValidationStage: {
+					SpoofGuardTable,
+				},
+				binding.ConntrackStateStage: {
+					ConntrackTable,
+					ConntrackStateTable,
+				},
+				binding.RoutingStage: {
+					L3ForwardingTable,
+					L3DecTTLTable,
+				},
+				binding.SwitchingStage: {
+					L2ForwardingCalcTable,
+				},
+				binding.ConntrackStage: {
+					ConntrackCommitTable,
+				},
+				binding.OutputStage: {
+					L2ForwardingOutTable,
+				},
+			},
+		}
+		for _, ipProtocol := range c.ipProtocols {
+			if ipProtocol == binding.ProtocolIPv6 {
+				template.stageTables[binding.ValidationStage] = append(template.stageTables[binding.ValidationStage],
```

Comment:
nit: maybe the `featureTemplate` struct should have an `addTable` method, so you could write `template.addTable(binding.ValidationStage, IPv6Table)` here

Reply:
Thanks, much better with method `addTable`